kernel/kernel-api/src/main/java/io/delta/kernel/TransactionBuilder.java

```diff
@@ -34,11 +34,26 @@
 @Evolving
 public interface TransactionBuilder {
   /**
-   * Set the schema of the table when creating a new table.
+   * Set the schema of the table. If setting the schema on an existing table for a schema evolution,
+   * then column mapping must be enabled. The possible schema evolutions supported include column
+   * additions, removals, renames, and moves. If a schema evolution is performed, implementations
+   * must perform the following validations:
+   *
+   * <ul>
+   *   <li>No duplicate columns are allowed
+   *   <li>Column names contain only valid characters
+   *   <li>Data types are supported
+   *   <li>No new non-nullable fields are added
+   *   <li>Physical column name consistency is preserved in the new schema
+   *   <li>ToDo: Nested IDs for array/map types are preserved in the new schema
```

Comment:
I've added this ToDo for now, after https://github.com/delta-io/delta/pull/4360 is in we can look at if this is even necessary in a follow on.